File: octopoints-flutter1/lib/db/DbSingleton.dart
import 'package:octopoints/db/database/OctopointsDb.dart';

class DbSingleton {
  DbSingleton._();
  static final DbSingleton _instance = DbSingleton._();

  factory DbSingleton() => _instance;

  static OctopointsDb? _db;

  OctopointsDb get db => _db!;

  Future<void> initDb() async {
    _db = await $FloorOctopointsDb.databaseBuilder('octopoints.db').build();
  }
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/DbSingleton.kt
package dev.tpavani.octopoints.db

import android.content.Context
import androidx.room.Room
import dev.tpavani.octopoints.db.database.OctopointsDb

class DbSingleton {

    companion object {
        private var db: OctopointsDb? = null

        fun getDb(): OctopointsDb{
            return db!!
        }

        fun initDb(context: Context) {
            db = Room.databaseBuilder(context, OctopointsDb::class.java, "octopoints.db").build()
        }
    }
}